### PR TITLE
Automatically convert $ to $$ when changing to string interpolation

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we wouldn't do anything about $ which would break the code.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala [hunk 1]
```diff
@@ -119,13 +119,30 @@ class StringActions(buffers: Buffers, trees: Trees) extends CodeAction {
...
-    val edits = List(
-      new l.TextEdit(startRange, "s")
-    )
+    val range = token.pos.toLSP
+    val start = range.getStart()
+    val dolarIndexes = token.value.indicesOf("$")
+    lazy val newlineIndexes = token.value.indicesOf("\n")
+    val dolarEdits = for (dolarIndex <- dolarIndexes) yield {
+      val newlinesBeforeDolar = newlineIndexes.takeWhile(_ < dolarIndex)
+      val countLines = newlinesBeforeDolar.size
```
Comment (kpodsiad) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala:130]: You can inline this.

**Review by kpodsiad — APPROVED** [review]

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala (hunk 1 above)
Comment (kpodsiad) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala:130]: You can inline `countLines`

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala [hunk 2]
```diff
@@ -119,13 +119,30 @@ class StringActions(buffers: Buffers, trees: Trees) extends CodeAction {
...
+      token: Token.Constant.String
   ): l.CodeAction = {
-    val startRange = new l.Range(range.getStart, range.getStart)
-
-    val edits = List(
-      new l.TextEdit(startRange, "s")
-    )
+    val range = token.pos.toLSP
+    val start = range.getStart()
+    val dolarIndexes = token.value.indicesOf("$")
```
Comment (Giggiux) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala:126]: Nitpicking: `dollar` instead of `dolar`

**Review by Giggiux — APPROVED** [review]
👍

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala (hunk 2 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala]: Fixed!

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala (hunk 1 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/StringActions.scala]: Done

**Review by kpodsiad — APPROVED** [review]

**Pull request merged**